backends/dpdk/dpdkContext.h

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2021 Intel Corp.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#ifndef BACKENDS_DPDK_CONTEXT_H_
+#define BACKENDS_DPDK_CONTEXT_H_
+
+#include "dpdkProgramStructure.h"
+#include "options.h"
+#include "lib/nullstream.h"
+
+/**
+Passes defined in this file are used for generating Context JSON output for DPDK
+Context JSON is a JSON file used by the control plane software for manipulating tables and
+actions. It contains all relevant information regarding the tables and actions.
+The context JSON is based on the JSON Schema defined in DPDK_context_schema.json.
+*/
+
+namespace DPDK {
+
+/* This structure holds table attributes required for context JSON which are not
+    part of P4Table */
+struct TableAttributes {
```

Comment:
Could you just have a method in this struct that converts it to JSON?
Then you could do the json generation by calling a few methods here and there and then serializing the resulting JSON object. It would be much nicer to read.

Reply:
I have updated the code use lib/json.h to create JSON object and serializing it